src/lib/comms/sol-coap.c

```diff
@@ -1746,40 +1746,30 @@ network_event(void *data, const struct sol_network_link *link, enum sol_network_
 }
 
 static struct sol_coap_server *
-sol_coap_server_new_full(enum sol_socket_type type, const struct sol_network_link_addr *servaddr)
+sol_coap_server_new_full(struct sol_socket_default_options *options, const struct sol_network_link_addr *servaddr)
```

Comment:
Move this to take the pre-built socket, this will avoid much of the hassle you have here. Add another function that creates your socket like for iotivity/oic needs (at sol-oic)